sdk/core/System.BinaryData/src/BinaryData.cs

```diff
@@ -0,0 +1,260 @@
+        /// <param name="jsonSerializable">The data to use.</param>
+        /// <param name="options">The <see cref="JsonSerializerOptions"/> to use when serializing to Json.</param>
+        ///
+        /// <returns>A <see cref="BinaryData"/> instance.</returns>
+        [Diagnostics.CodeAnalysis.SuppressMessage("Usage", "AZC0014:Avoid using banned types in public API", Justification = "<Pending>")]
+        public static BinaryData FromObjectAsJson<T>(
+            T jsonSerializable,
+            JsonSerializerOptions? options = default)
+        {
+            using var memoryStream = new MemoryStream();
+            var buffer = JsonSerializer.SerializeToUtf8Bytes(jsonSerializable, typeof(T), options);
+            memoryStream.Write(buffer, 0, buffer.Length);
+            return new BinaryData(memoryStream.GetBuffer().AsMemory(0, (int) memoryStream.Position));
+        }
+
+        /// <summary>
+        /// Converts the <see cref="BinaryData"/> to a string using UTF-8.
+        /// </summary>
+        /// <returns>The string representation of the <see cref="BinaryData"/> using UTF-8
+        /// to decode the bytes.</returns>
+        public override string ToString()
+        {
+            if (MemoryMarshal.TryGetArray(
+                Bytes,
+                out ArraySegment<byte> data))
+            {
+                return s_encoding.GetString(data.Array, data.Offset, data.Count);
+            }
+            return s_encoding.GetString(Bytes.ToArray());
+        }
+
+        /// <summary>
+        /// Converts the <see cref="BinaryData"/> to a read-only stream.
+        /// </summary>
+        /// <returns>A stream representing the data.</returns>
+        public Stream ToStream() =>
+            new ReadOnlyMemoryStream(Bytes);
+
+        /// <summary>
+        /// Gets the <see cref="BinaryData"/> as <see cref="ReadOnlyMemory{Byte}"/>.
+        /// </summary>
+        /// <returns><see cref="BinaryData"/> as bytes.</returns>
+        public ReadOnlyMemory<byte> ToBytes() =>
+            Bytes;
+
+        /// <summary>
+        /// Converts the <see cref="BinaryData"/> to the specified type using
+        /// <see cref="JsonSerializer"/>.
+        /// </summary>
+        /// <typeparam name="T">The type that the data should be
+        /// converted to.</typeparam>
+        /// <param name="options">The <see cref="JsonSerializerOptions"/> to use when serializing to Json.</param>
+        /// <returns>The data converted to the specified type.</returns>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+        public T ToObjectFromJson<T>(JsonSerializerOptions? options = default)
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+        {
+            using var memoryStream = new MemoryStream();
+            ToStream().CopyTo(memoryStream);
+            return (T) JsonSerializer.Deserialize(memoryStream.ToArray(), typeof(T), options);
+
```

Comment:
Why is there a second stream being created?  ToStream returns a new value each time, and JsonSerializer.Deserialize isn't writing back into that stream. 

Actually, why is a first stream being created?


```suggestion
            return JsonSerializer.Deserialize<T>(_bytes.Span, options);
```